Clarified roles versus permissions usage

BeateRixen opened the pull request:
To clarify the inconsistent usage of the terms "role" and "permission" in the UI and API, I made the following changes:

- Added a note in
- - https://cumulocity.com/docs/standard-tenant/managing-permissions/
- - https://cumulocity.com/docs/microservice-sdk/general-aspects/#microservice-manifest
- - https://cumulocity.com/docs/microservice-sdk/general-aspects/#security
- - https://cumulocity.com/docs/microservice-sdk/rest/#microservice-development
- Checked throughout the documentation and corrected the usage of these terms in various cases.
- Renamed the section "Managing permissions" to "Managing permissions and roles" as this is actually what it is about.

The glossary entries for "role" and "permission" will be updated in a separate PR together with the API details.

BeateRixen committed:
Clarified usage of the terms roles and permissions

BeateRixen committed:
improved the notes

BeateRixen committed:
added one more sentence to a note

BeateRixen committed:
Update managing-permissions.md

BeateRixen committed:
small fixes after review

BeateRixen committed:
Merge branch 'develop' into roles-permissions

ben-c8y commented on content/microservice-sdk/general-aspects-bundle/microservice-manifest.md:
```diff
@@ -160,13 +160,13 @@ See below for detailed information about available settings.
 <td style="text-align:left">requiredRoles</td>
 <td style="text-align:left">String[ ]</td>
-<td style="text-align:left">List of permissions required by a microservice to work.<br/>Default: [ ] (no permissions)</td>
+<td style="text-align:left">List of permissions required by a microservice to work. <b>Important:</b> Be aware that despite of the term "role" in the field name the field expects a "permission" string (for example, ROLE_INVENTORY_ADMIN). <br/>Default: [ ] (no permissions)</td>
```
```suggestion
<td style="text-align:left">List of permissions required by a microservice to work. <b>Important:</b> Be aware that despite the term "role" in the field name, the field expects a "permission" string (for example, ROLE_INVENTORY_ADMIN). <br/>Default: [ ] (no permissions)</td>
```

I think "despite of" isn't very normal grammar, or at least doesn't read so clearly to me (also in a few other places)

ben-c8y commented on content/microservice-sdk/general-aspects-bundle/security.md:
```diff
@@ -135,13 +135,13 @@ There are three types of users:
-* Required roles: The roles that are predefined to allow access to {{< product-c8y-iot >}} REST APIs.
-For instance, if a microservice creates measurements using the service user, measurement admin role must be added as a required role of the application.
+* Required roles: The permissions that are predefined to allow access to {{< product-c8y-iot >}} REST APIs.
+For instance, if a microservice creates measurements using the service user, measurement ADMIN permission must be added as a required role/permission of the application.
```
```suggestion
For instance, if a microservice creates measurements using the service user, the measurement ADMIN permission must be added as a required role/permission of the application.
```

ben-c8y commented on content/microservice-sdk/general-aspects-bundle/security.md:
```diff
@@ -135,13 +135,13 @@ There are three types of users:
+These permissions can be assigned or revoked to the tenant platform users or groups using the Administration application.
 
-Custom roles must adhere to this name format in order to be shown in the UI:
+Custom roles/permissions must adhere to this name format in order to be shown in the UI:
```
actually I think this statement is meant to apply to permissions only - these are just permissions strings not roles

ben-c8y commented on content/microservice-sdk/general-aspects-bundle/security.md:
```diff
@@ -156,7 +156,11 @@ You can add them to the [application manifest](#microservice-manifest) in the `r
 <!-- TODO: add/describe a picture of "required roles" and "provided roles" showing a microservice as a block -->
 
-The roles are set in the [Microservice manifest](#microservice-manifest). For more details about users and roles, review the [User API](https://{{< domain-c8y >}}/api/core/#tag/User-API) in the {{< openapi >}}.
+The roles are set in the [Microservice manifest](#microservice-manifest). For more details about users and roles, refer to [Managing permissions and roles](/standard-tenant/managing-permissions) or to the [User API](https://{{< domain-c8y >}}/api/core/#tag/User-API) in the {{< openapi >}}.
```
I _think_ this is still referring to permissions not roles

BeateRixen replied:
```suggestion
The permissions are set in the [Microservice manifest](#microservice-manifest). For more details about users, permissions, and roles, refer to [Managing permissions and roles](/standard-tenant/managing-permissions) or to the [User API](https://{{< domain-c8y >}}/api/core/#tag/User-API) in the {{< openapi >}}.
```

ben-c8y commented on content/microservice-sdk/general-aspects-bundle/security.md:
```diff
@@ -156,7 +156,11 @@ You can add them to the [application manifest](#microservice-manifest) in the `r
+The roles are set in the [Microservice manifest](#microservice-manifest). For more details about users and roles, refer to [Managing permissions and roles](/standard-tenant/managing-permissions) or to the [User API](https://{{< domain-c8y >}}/api/core/#tag/User-API) in the {{< openapi >}}.
+
+{{< c8y-admon-important >}}
+Be aware that the term "role" might be misleading here as the the `roles`field and `requiredRoles` field expect a "permission" string (for example, ROLE_INVENTORY_ADMIN). See also the glossary for the usage of the terms [permission](/glossary/#permission) and [role](/glossary/#role) in the Cumulocity context.
```
```suggestion
Be aware that the term "role" might be misleading here as the the `roles` field and `requiredRoles` field expect a "permission" string (for example, ROLE_INVENTORY_ADMIN). See also the glossary for the usage of the terms [permission](/glossary/#permission) and [role](/glossary/#role) in the Cumulocity context.
```

whitespace typo

ben-c8y commented on content/microservice-sdk/java-bundle/ip-tracker-microservice.md:
```diff
@@ -57,7 +57,7 @@ Finally, add the following dependency:
 In your _cumulocity.json_ file:
 
-1. Add the required roles to be able to create events and alarms.
+1. Add the required roles (permissions) to be able to create events and alarms.
```
```suggestion
1. Add the required permissions (using the `requiredRoles` field) to be able to create events and alarms.
```

I think this is what's meant?

ben-c8y commented on content/microservice-sdk/rest-bundle/microservice-development.md:
```diff
@@ -30,9 +30,13 @@ A success response consists of a 201 status and a location header similar to `<H
 * `roles` - A list of microservice permissions. If the microservice exposes an own REST API, it can be secured with an own set of permissions, for example, a SMS microservice would require `SMS_ADMIN` permission to send SMS messages. These permissions become available in the tenant after microservice subscription. Afterwards, an admin user can grant such permission to  another user that wants to  ...
 
+{{< c8y-admon-important >}}
+Be aware that the term "role" might be misleading here as the `roles`field and the `requiredRoles` field expect a "permission" string (for example, ROLE_INVENTORY_ADMIN). Also see the glossary for the usage of the terms [permission](/glossary/#permission) and [role](/glossary/#role) in the Cumulocity context.
```
```suggestion
Be aware that the term "role" might be misleading here as the `roles` field and the `requiredRoles` field expect a "permission" string (for example, `ROLE_INVENTORY_ADMIN`). Also see the glossary for the usage of the terms [permission](/glossary/#permission) and [role](/glossary/#role) in the Cumulocity context.
```

typo

ben-c8y commented on content/microservice-sdk/rest-bundle/microservice-development.md:
```diff
@@ -30,9 +30,13 @@ A success response consists of a 201 status and a location header similar to `<H
 
-* `requiredRoles` - A list of {{< product-c8y-iot >}} permissions the microservice user needs in order to get data from {{< product-c8y-iot >}}, for example, if the microservice creates a managed object, one of the required roles shall be `ROLE_INVENTORY_ADMIN`.
+* `requiredRoles` - A list of {{< product-c8y-iot >}} permissions the microservice user needs in order to get data from {{< product-c8y-iot >}}, for example, if the microservice creates a managed object, one of the required permissions shall be `ROLE_INVENTORY_ADMIN`.
 * `roles` - A list of microservice permissions. If the microservice exposes an own REST API, it can be secured with an own set of permissions, for example, a SMS microservice would require `SMS_ADMIN` permission to send SMS messages. These permissions become available in the tenant after microservice subscription. Afterwards, an admin user can grant such permission to  another user that wants to  ...
```
```suggestion
* `roles` - A list of microservice permissions defined by this microservice. If the microservice exposes its own REST API, it can be secured with its own set of permissions, for example, an SMS microservice would require `SMS_ADMIN` permission to send SMS messages. These permissions become available in the tenant after microservice subscription. Afterwards, an admin user can grant such permission to  another user that wants to send SMS messages via the {{< product-c8y-iot >}} platform.
```

Clarified meaning (please check this is correct) and improved grammar a bit in this existing sentence

ben-c8y commented on content/standard-tenant/managing-permissions.md:
```diff
@@ -29,7 +35,7 @@ ROLES & PERMISSIONS:
 * To create new roles with available (owned) permissions: CREATE and ADMIN permission.
 
-The above permissions can be used to create roles for robust user management. Every new tenant have specified typical roles by default:
+The above permissions can be used to create roles for robust user management. Every new tenant has specified typical roles by default:
```
```suggestion
The above permissions can be used to create roles for robust user management. Every new tenant has these roles by default:
```

Grammar improvement?

ben-c8y reviewed (commented):
This looks great Beate, and I think will make the topic of permissions/roles far clearer for our users

I added some comments, mostly small cosmetic things

BeateRixen committed:
Update content/microservice-sdk/general-aspects-bundle/microservice-manifest.md

Co-authored-by: Ben Spiller <84027153+ben-c8y@users.noreply.github.com>

BeateRixen commented on content/microservice-sdk/general-aspects-bundle/microservice-manifest.md:
```diff
@@ -160,13 +160,13 @@ See below for detailed information about available settings.
 <td style="text-align:left">roles</td>
 <td style="text-align:left">String[ ]</td>
-<td style="text-align:left">Roles provided by the microservice. <br>Default: [ ] (empty list)</td>
+<td style="text-align:left">Roles provided by the microservice. <b>Important:</b> Be aware that despite of the term "role" in the field name the field expects a "permission" string (for example, ROLE_INVENTORY_ADMIN). <br>Default: [ ] (empty list)</td>
```
```suggestion
<td style="text-align:left">Roles provided by the microservice. <b>Important:</b> Be aware that despite the use of the term "role" in the field name, the field expects a "permission" string (for example, ROLE_INVENTORY_ADMIN). <br>Default: [ ] (empty list)</td>
```

BeateRixen committed:
Update content/microservice-sdk/general-aspects-bundle/microservice-manifest.md

BeateRixen commented on content/microservice-sdk/general-aspects-bundle/microservice-manifest.md:
```diff
@@ -160,13 +160,13 @@ See below for detailed information about available settings.
 <td style="text-align:left">requiredRoles</td>
 <td style="text-align:left">String[ ]</td>
-<td style="text-align:left">List of permissions required by a microservice to work.<br/>Default: [ ] (no permissions)</td>
+<td style="text-align:left">List of permissions required by a microservice to work. <b>Important:</b> Be aware that despite the term "role" in the field name, the field expects a "permission" string (for example, ROLE_INVENTORY_ADMIN). <br/>Default: [ ] (no permissions)</td>
```
```suggestion
<td style="text-align:left">List of permissions required by a microservice to work. <b>Important:</b> Be aware that despite the use of term "role" in the field name, the field expects a "permission" string (for example, ROLE_INVENTORY_ADMIN). <br/>Default: [ ] (no permissions)</td>
```

BeateRixen committed:
Update content/microservice-sdk/general-aspects-bundle/microservice-manifest.md

BeateRixen committed:
Apply suggestions from code review

Co-authored-by: Ben Spiller <84027153+ben-c8y@users.noreply.github.com>

BeateRixen commented on content/microservice-sdk/general-aspects-bundle/security.md:
```diff
@@ -135,13 +135,13 @@ There are three types of users:
+* Required roles: The permissions that are predefined to allow access to {{< product-c8y-iot >}} REST APIs.
+For instance, if a microservice creates measurements using the service user, the measurement ADMIN permission must be added as a required role/permission of the application.
 Required roles are added to the service users.
 * Roles: The custom roles provided to tenant platform users by the microservice developer.
```
```suggestion
* Roles: The custom permissions provided to tenant platform users by the microservice developer.
```

BeateRixen commented on content/microservice-sdk/general-aspects-bundle/security.md:
```diff
@@ -135,13 +135,13 @@ There are three types of users:
+These permissions can be assigned or revoked to the tenant platform users or groups using the Administration application.
 
-Custom roles must adhere to this name format in order to be shown in the UI:
+Custom roles/permissions must adhere to this name format in order to be shown in the UI:
```
```suggestion
Custom rpermissions must adhere to this name format to be shown in the UI:
```

BeateRixen committed:
Apply suggestions from code review

BeateRixen commented on content/microservice-sdk/general-aspects-bundle/security.md:
```diff
@@ -135,13 +135,13 @@ There are three types of users:
+These permissions can be assigned or revoked to the tenant platform users or groups using the Administration application.
 
-Custom roles must adhere to this name format in order to be shown in the UI:
+Custom rpermissions must adhere to this name format to be shown in the UI:
```
```suggestion
Custom permissions must adhere to this name format to be shown in the UI:
```

BeateRixen committed:
Update content/microservice-sdk/general-aspects-bundle/security.md

BeateRixen committed:
Update content/standard-tenant/managing-permissions.md

Co-authored-by: Ben Spiller <84027153+ben-c8y@users.noreply.github.com>

BeateRixen commented:
> This looks great Beate, and I think will make the topic of permissions/roles far clearer for our users
> 
> I added some comments, mostly small cosmetic things

@ben-c8y I addressed your comments. Please approve.

beabehnisch commented on content/change-logs/platform-services/cumulocity-undefined-pre-announce-hz-changes.md:
```diff
@@ -19,8 +19,8 @@ the default on  https://eu-latest.cumulocity.com/ and available on all other ins
 
-Enabling the new implementation is on a per tenant basis and requires the `ROLE_TENANT_MANAGEMENT_ADMIN` role and use of
-the [Feature Toggles REST API](https://cumulocity.com/api/core/#operation/setCurrentTenantFeatureToggleValue), i.e.:
+Enabling the new implementation is on a per tenant basis and requires the `ROLE_TENANT_MANAGEMENT_ADMIN` permission and use of
```
As this is not the API docu, I would recommend a more readable and UI centric way to refer to permissions by splitting up its structure (scope + access level= permission). 

Optionally we can (at least once in a section) mention "requires your role to include ... xyz permissions" to make clear how permissions are actually assigned. 

The API string could be much more hidden, i. e. in brackets for single permissions mentioned - or in case of many permissions mentioned as a reference/translation table targeting API users. 

```suggestion
Enabling the new implementation is on a per tenant basis and requires your role to include ADMIN permission for "Tenant management" (API string = `ROLE_TENANT_MANAGEMENT_ADMIN`) and use of
```

beabehnisch commented on content/cloud-remote-access/setting-up-cloud-remote-access-bundle/endpoints.md:
```diff
@@ -13,7 +13,7 @@ The "endpoint" is the IP address and port of the VNC, SSH or Telnet server runni
 
 {{< c8y-admon-info >}}
-To be able to configure an endpoint, you need ADMIN permission for "Remote access" and "Device control". To read data, a READ permission is sufficient. For more information on permissions, refer to [Managing permissions](/standard-tenant/managing-permissions/).
+To be able to configure an endpoint, you need ADMIN permission for "Remote access" and "Device control". To read data, a READ permission is sufficient. For more information on permissions, refer to [Managing permissions and roles](/standard-tenant/managing-permissions/).
```
Just to consider incl. "Your role must include" here as well 


```suggestion
To be able to configure an endpoint, you must have a role that includes ADMIN permission for "Remote access" and "Device control". To read data, READ permission is sufficient. For more information on permissions, refer to [Managing permissions and roles](/standard-tenant/managing-permissions/).
```

beabehnisch commented on content/concepts/security-bundle/access-control.md:
```diff
@@ -19,7 +19,7 @@ Permissions define explicitly what functionality can be executed by a user.
 {{< product-c8y-iot >}} distinguishes read permissions and administration permissions. Read permissions enable users to read data. Administration permissions enable users to create, update and delete data. Read and administration permissions are separately available for the different types of data in {{< product-c8y-iot >}}. For example, there are read permissions for inventory data, measurements ...
 
-To manage permissions more easily, they are grouped into so-called "roles". Every user can be associated with a number of roles, adding up permissions of the user.
+To manage permissions more easily, they are grouped into roles. Every user can be associated with a number of roles, adding up permissions of the user.
```
adding up permission reads a bit clunky and I would try to stick to "assigning" roles as it is more actionable – just a suggestion to make it easier to read.

```suggestion
To manage permissions more easily, they are grouped into roles. When you assign one or more roles to a user, they gain the combined permissions of all these roles.
```

beabehnisch commented on content/datahub/setting-up-datahub-bundle/defining-permissions.md:
```diff
@@ -4,7 +4,7 @@ title: Defining Cumulocity DataHub permissions and roles
 ---
 
-Dedicated permissions define what a user is allowed to do in {{< product-c8y-iot >}} DataHub. To ease assigning permissions to users, permissions are grouped in roles. During deployment of the {{< product-c8y-iot >}} DataHub applications the corresponding permissions as well as roles are created. If a role with the same name already exists, no new role will be created. The same holds for permissi ...
+Dedicated permissions define what a user is allowed to do in {{< product-c8y-iot >}} DataHub. To ease assigning permissions to users, permissions are grouped into roles. During deployment of the {{< product-c8y-iot >}} DataHub applications the corresponding permissions as well as roles are created. If a role with the same name already exists, no new role will be created. The same holds for permis ...
```
```suggestion
Dedicated permissions define what a user is allowed to do in {{< product-c8y-iot >}} DataHub. You assign these permissions to users by using roles, which group permission into bundles. During deployment of the {{< product-c8y-iot >}} DataHub applications the corresponding permissions as well as roles are created. If a role with the same name already exists, no new role will be created. The same holds for permissions.
```

beabehnisch commented on content/device-certificate-authentication/certificate-authority.md:
```diff
@@ -104,7 +104,7 @@ The following response is returned:
 
 {{< c8y-admon-info >}}
-In order to call `/certificate-authority` one of the following roles is required: ROLE_TENANT_MANAGEMENT_ADMIN or ROLE_TENANT_ADMIN, otherwise an HTTP response 403 will be returned.
+In order to call `/certificate-authority` one of the following permissions is required: ROLE_TENANT_MANAGEMENT_ADMIN or ROLE_TENANT_ADMIN, otherwise an HTTP response 403 will be returned.
```
just a comma:

```suggestion
In order to call `/certificate-authority`, one of the following permissions is required: ROLE_TENANT_MANAGEMENT_ADMIN or ROLE_TENANT_ADMIN, otherwise an HTTP response 403 will be returned.
```

... or even here we could improve readability and focus on the permission pattern (and only secondary on the string), as we are still not in the API docu :)

 ```suggestion
To call `/certificate-authority`, your role must include ADMIN permission for "Tenant" (API string = `ROLE_TENANT_ADMIN`) or "Tenant management" (API string = `ROLE_TENANT_MANAGEMENT_ADMIN`). Without this, the request will fail (HTTP 403 error).
```

beabehnisch commented on content/device-integration/lpwan-custom-codec-bundle/using-the-lpwan-custom-codec-library.md:
````diff
@@ -75,7 +75,7 @@ To create a custom codec microservice using this library, do the following:
         ```
 
-4. Add the following roles as `requiredRoles` in the microservice manifest file `cumulocity.json`:
+4. Add the following permissions as `requiredRoles` in the microservice manifest file `cumulocity.json`:
````
while this works and is much more correct, it still sounds odd. 
An alternative to avoid it could be:

```suggestion
4. Add the following permissions in the microservice manifest file `cumulocity.json`:
```
then after the JSON code, maybe a Note: 
While the field in the manifest is named requiredRoles, you are adding the list of specific permission strings required.

BeateRixen replied:
```suggestion
4. Add the following permissions in the microservice manifest file `cumulocity.json`. Note that the respective field in the manifest is called `requiredRoles`, but what you actually add is a list of required permissions strings.
```

beabehnisch commented on content/glossary/a.md:
```diff
@@ -10,7 +10,7 @@ _build:
 ### Administration application {#administration-application}
 
-The Administration application is a default {{< product-c8y-iot >}} application that serves as the central management console for platform administrators. It is used to govern a [tenant](#tenant) by managing [users](#user), roles, and [permissions](#permission), subscribing to and managing [applications](#application) and [microservices](#microservice), and configuring tenant-level settings such  ...
+The Administration application is a default {{< product-c8y-iot >}} application that serves as the central management console for platform administrators. It is used to govern a [tenant](#tenant) by managing [users](#user), [roles](#role), and [permissions](#permission), subscribing to and managing [applications](#application) and [microservices](#microservice), and configuring tenant-level setti ...
```
thinking about the new billing app called Cumulocity Console, we can erase this term here already:

```suggestion
The Administration application is a default {{< product-c8y-iot >}} application and the central place for platform administrators to manage a tenant. It is used to govern a [tenant](#tenant) by managing [users](#user), [roles](#role), and [permissions](#permission), subscribing to and managing [applications](#application) and [microservices](#microservice), and configuring tenant-level settings such as retention rules, custom properties, and [branding](#branding).   
```

beabehnisch commented on content/microservice-sdk/general-aspects-bundle/security.md:
```diff
@@ -135,13 +135,13 @@ There are three types of users:
-* Required roles: The roles that are predefined to allow access to {{< product-c8y-iot >}} REST APIs.
-For instance, if a microservice creates measurements using the service user, measurement admin role must be added as a required role of the application.
+* Required roles: The permissions that are predefined to allow access to {{< product-c8y-iot >}} REST APIs.
+For instance, if a microservice creates measurements using the service user, the measurement ADMIN permission must be added as a required role/permission of the application.
```
#```suggestion
For instance, if a microservice creates measurements using the service user, ADMIN permission for "Measurements" must be added as a required role/permission of the application.
```

BeateRixen replied:
```suggestion
For instance, if a microservice creates measurements using the service user, the ADMIN permission for "Measurement" must be added as a required permission of the application.
```

beabehnisch commented on content/microservice-sdk/java-bundle/services-platform-sms-api.md:
```diff
@@ -30,10 +30,10 @@ SmsMessagingApi smsMessagingApi = platform.getSmsMessagingApi();
 
-To use the SMS messaging API, the user must have the required roles SMS_ADMIN and SMS_READ for sending and receiving messages respectively.
-Refer to [Managing permissions](/standard-tenant/managing-permissions/) for more information.
+To use the SMS messaging API, the user must have the required permissions SMS_ADMIN and SMS_READ for sending and receiving messages respectively.
```
```suggestion
To use the SMS messaging API, the user must have the ADMIN and READ permission for "SMS" (SMS_ADMIN and SMS_READ) for sending and receiving messages respectively.
```

beabehnisch commented on content/standard-tenant/changing-settings-bundle/localization.md:
```diff
@@ -21,7 +21,7 @@ Using the **Localization** functionality you can add custom translations for exi
 - To add/update/delete localization identifiers: ADMIN permission for permission type "Application management"
 
-- Your user must have a role with READ permission for the permission type "Application management". See [Managing permissions](/standard-tenant/managing-permissions/) for more information.
+- Your user must have a role with READ permission for the permission type "Application management". See [Managing permissions and roles](/standard-tenant/managing-permissions/) for more information.
```
we can shorten it

```suggestion
- Your user must have a role with READ permission for "Application management". See [Managing permissions and roles](/standard-tenant/managing-permissions/) for more information.
```

beabehnisch commented on content/standard-tenant/ecosystem-bundle/monitoring-microservices.md:
```diff
@@ -14,7 +14,7 @@ The status of the microservice can be checked in the **Status** tab of the respe
 <img src="/images/users-guide/Administration/admin-microservice-status.png" alt="Microservice status" style="max-width: 100%">
 
-To view the status you need the following permissions: role Application management READ and role Inventory READ.
+To view the status you need the following permissions: READ permission for Application management and READ permission for Inventory.
```
```suggestion
To view the status you need the following permissions: READ permission for "Application management" and "Inventory".
```

beabehnisch commented on content/standard-tenant/managing-permissions.md:
```diff
@@ -10,7 +10,7 @@ sector:
 - label: managing-permissions
   title: Managing permissions
-  content: "Permissions define what a user is allowed to do in Cumulocity applications. To manage permissions more easily, they are grouped into so-called 'roles'. Every user can be associated with a number of roles, adding up permissions of the user.
+  content: "Permissions define what a user is allowed to do in Cumulocity applications. To manage permissions more easily, they are grouped into global and inventory roles. Every user can be associated with a number of roles, adding up permissions of the user.
```
trying to use 'assign roles' again and avoid "adding up" (same suggestion as in content/concepts/security-bundle/access-control.md)

```suggestion
  content: "Permissions define what a user is allowed to do in Cumulocity applications. To manage permissions more easily, they are grouped into global and inventory roles. When you assign one or more roles to a user, they gain the combined permissions of all these roles.
```

beabehnisch commented on content/streaming-analytics/analytics-customization-bundle/control-access.md:
```diff
@@ -67,7 +67,7 @@ curl --user username -X POST -H 'Content-Type: application/json' -d '{"name":"fe
 By default, all users can see the same set of pages (according to the limitations above).
 You can also restrict the visibility of the pages to only users who have the permission ROLE_ANALYTICSBUILDER_READ or ROLE_EPLAPPS_READ,
-which can be assigned directly to users or via groups, see also [Managing permissions](/standard-tenant/managing-permissions/).
+which can be assigned directly to users or via groups, see also [Managing permissions and roles](/standard-tenant/managing-permissions/).
```
@ben-c8y How can I assign READ permission for "Analytics Builder" or "EPL apps" **directly** to a user? At least via UI I woudn't know how and assume I also have to assign a role that contains these. Is this a special permission somehow or targeting only API users?

ben-c8y replied:
@beabehnisch 
yes you're right. There's no weird magic here - these are permissions for which you'd crreate a role and give the role to the users/groups you need.

I suggest just removing "which can be assigned directly to users or via groups" and going straight to the see also link which tells people how permissions are assigned

beabehnisch reviewed (approved):
These are all important and great improvements! 
Just added some suggestions and a few tiny issues. Nothing critical.

neuerburg reviewed (approved):
Good changes!

BeateRixen committed:
Update content/standard-tenant/managing-permissions.md

Co-authored-by: Beatrice Behnisch <52067789+beabehnisch@users.noreply.github.com>

BeateRixen committed:
Apply suggestions from code review

Changes from review

Co-authored-by: Beatrice Behnisch <52067789+beabehnisch@users.noreply.github.com>

BeateRixen commented on content/streaming-analytics/analytics-customization-bundle/control-access.md:
```diff
@@ -67,7 +67,7 @@ curl --user username -X POST -H 'Content-Type: application/json' -d '{"name":"fe
 By default, all users can see the same set of pages (according to the limitations above).
 You can also restrict the visibility of the pages to only users who have the permission ROLE_ANALYTICSBUILDER_READ or ROLE_EPLAPPS_READ,
-which can be assigned directly to users or via groups, see also [Managing permissions](/standard-tenant/managing-permissions/).
+which can be assigned directly to users or via groups, see also [Managing permissions and roles](/standard-tenant/managing-permissions/).
```
```suggestion
see also [Managing permissions and roles](/standard-tenant/managing-permissions/).
```

BeateRixen committed:
Update content/streaming-analytics/analytics-customization-bundle/control-access.md
